### PR TITLE
[New Rule] New BBR Rules - Part 5

### PR DESCRIPTION
**Pull request by w0rk3r**
## Issues

https://github.com/elastic/ia-trade-team/issues/158

## Summary

Adds the following New BBR Rules:

* Service Disabled via Registry Modification
* Service Path Modification
* Service Path Modification via sc.exe
* Potential Outgoing RDP Connection by Unusual Process
* Netsh Helper DLL
* Shortcut File Written or Modified on Startup Folder
* Werfault ReflectDebugger Persistence

Adjusts the following rules:

* Windows Account or Group Discovery

### CODE REVIEW

File: rules_building_block/lateral_movement_rdp_conn_unusual_process.toml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+tags = ["Domain: Endpoint", "OS: Windows", "Use Case: Threat Detection", "Tactic: Lateral Movement"]
+timestamp_override = "event.ingested"
+building_block_type = "default"
+type = "eql"
+
+query = '''
+network where host.os.type == "windows" and
+  event.action : "connection_attempted" and destination.port == 3389 and
+  not process.executable : "?:\\Windows\\System32\\mstsc.exe" and
+  (destination.ip != "::1" and destination.ip != "127.0.0.1")
```
Comment (Aegrah) [inline rules_building_block/lateral_movement_rdp_conn_unusual_process.toml:34]:
```suggestion
  destination.ip != "::1" and destination.ip != "127.0.0.1"
```

File: rules_building_block/lateral_movement_rdp_conn_unusual_process.toml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+from = "now-119m"
+interval = "60m"
+index = ["logs-endpoint.events.*"]
+language = "eql"
+license = "Elastic License v2"
+name = "Potential Outgoing RDP Connection by Unusual Process"
+risk_score = 21
+rule_id = "8e39f54e-910b-4adb-a87e-494fbba5fb65"
+severity = "low"
+tags = ["Domain: Endpoint", "OS: Windows", "Use Case: Threat Detection", "Tactic: Lateral Movement"]
```
Comment (Aegrah) [inline rules_building_block/lateral_movement_rdp_conn_unusual_process.toml:25]:
```suggestion
tags = ["Domain: Endpoint", "OS: Windows", "Use Case: Threat Detection", "Tactic: Lateral Movement", "Data Source: Elastic Defend"]
```

File: rules_building_block/lateral_movement_rdp_conn_unusual_process.toml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+maturity = "production"
+min_stack_comments = "New fields added: required_fields, related_integrations, setup"
+min_stack_version = "8.3.0"
+updated_date = "2023/08/29"
+
+[rule]
+author = ["Elastic"]
+description = """
+Adversaries may attempt to connect to a remote system over Windows Remote Desktop Protocol (RDP) to achieve lateral
+movement. Sometimes, an adversary may avoid using the Microsoft Terminal Services Client (mstsc.exe) binary to establish
```
Comment (Aegrah) [inline rules_building_block/lateral_movement_rdp_conn_unusual_process.toml:13]:
```suggestion
movement. Adversaries may avoid using the Microsoft Terminal Services Client (mstsc.exe) binary to establish
```

File: rules_building_block/lateral_movement_rdp_conn_unusual_process.toml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+rule_id = "8e39f54e-910b-4adb-a87e-494fbba5fb65"
+severity = "low"
+tags = ["Domain: Endpoint", "OS: Windows", "Use Case: Threat Detection", "Tactic: Lateral Movement"]
+timestamp_override = "event.ingested"
+building_block_type = "default"
+type = "eql"
+
+query = '''
+network where host.os.type == "windows" and
+  event.action : "connection_attempted" and destination.port == 3389 and
```
Comment (Aegrah) [inline rules_building_block/lateral_movement_rdp_conn_unusual_process.toml:32]:
```suggestion
  event.action == "connection_attempted" and destination.port == 3389 and
```

File: rules_building_block/persistence_netsh_helper_dll.toml [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+risk_score = 21
+rule_id = "b0638186-4f12-48ac-83d2-47e686d08e82"
+severity = "low"
+tags = ["Domain: Endpoint", "OS: Windows", "Use Case: Threat Detection", "Tactic: Persistence", "Data Source: Elastic Endgame", "Rule Type: BBR"]
+timestamp_override = "event.ingested"
+building_block_type = "default"
+type = "eql"
+
+query = '''
+registry where event.type : "change" and
```
Comment (Aegrah) [inline rules_building_block/persistence_netsh_helper_dll.toml:31]:
```suggestion
registry where event.type == "change" and
```

File: rules_building_block/persistence_werfault_reflectdebugger.toml [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+risk_score = 21
+rule_id = "205b52c4-9c28-4af4-8979-935f3278d61a"
+severity = "low"
+tags = ["Domain: Endpoint", "OS: Windows", "Use Case: Threat Detection", "Tactic: Persistence", "Data Source: Elastic Endgame", "Rule Type: BBR"]
+timestamp_override = "event.ingested"
+building_block_type = "default"
+type = "eql"
+
+query = '''
+registry where event.type : "change" and
```
Comment (Aegrah) [inline rules_building_block/persistence_werfault_reflectdebugger.toml:30]:
```suggestion
registry where event.type == "change" and
```

File: rules_building_block/defense_evasion_service_disabled_registry.toml [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+query = '''
+registry where host.os.type == "windows" and event.type == "change" and
+  registry.path : (
+    "HKLM\\SYSTEM\\*ControlSet*\\Services\\*\\Start",
+    "\\REGISTRY\\MACHINE\\SYSTEM\\*ControlSet*\\Services\\*\\Start"
+  ) and registry.data.strings : ("3", "4") and
+  not 
+    (
+      process.name: "services.exe" and user.id : "S-1-5-18"
```
Comment (Aegrah) [inline rules_building_block/defense_evasion_service_disabled_registry.toml:44]:
```suggestion
      process.name : "services.exe" and user.id : "S-1-5-18"
```

File: rules_building_block/defense_evasion_service_path_registry.toml [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+building_block_type = "default"
+type = "eql"
+
+query = '''
+registry where host.os.type == "windows" and event.type == "change" and
+  registry.path : (
+    "HKLM\\SYSTEM\\*ControlSet*\\Services\\*\\ImagePath",
+    "\\REGISTRY\\MACHINE\\SYSTEM\\*ControlSet*\\Services\\*\\ImagePath"
+  ) and not (
+    process.executable: (
```
Comment (Aegrah) [inline rules_building_block/defense_evasion_service_path_registry.toml:42]:
```suggestion
    process.executable : (
```

**Review by Aegrah — COMMENTED** [review]
Nice work! Just some nits, and a question:

In the EndGame supported rules we add `"Data Source: Elastic Endgame"`, but should we additionally add `"Data Source: Elastic Defend"`?

**Comment by w0rk3r** [comment]
> In the EndGame supported rules we add "Data Source: Elastic Endgame", but should we additionally add "Data Source: Elastic Defend" as they are also leveraging this data source?

Yup

File: rules_building_block/defense_evasion_service_disabled_registry.toml [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+name = "Service Disabled via Registry Modification"
+risk_score = 21
+rule_id = "75dcb176-a575-4e33-a020-4a52aaa1b593"
+severity = "low"
+tags = [
+    "Domain: Endpoint",
+    "OS: Windows",
+    "Use Case: Threat Detection",
+    "Tactic: Defense Evasion",
+    "Data Source: Elastic Endgame",
```
Comment (w0rk3r) [inline rules_building_block/defense_evasion_service_disabled_registry.toml:29]:
```suggestion
    "Data Source: Elastic Endgame",
    "Data Source: Elastic Defend",
```

File: rules_building_block/defense_evasion_service_path_registry.toml [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+name = "Service Path Modification"
+risk_score = 21
+rule_id = "f243fe39-83a4-46f3-a3b6-707557a102df"
+severity = "low"
+tags = [
+    "Domain: Endpoint",
+    "OS: Windows",
+    "Use Case: Threat Detection",
+    "Tactic: Defense Evasion",
+    "Data Source: Elastic Endgame",
```
Comment (w0rk3r) [inline rules_building_block/defense_evasion_service_path_registry.toml:29]:
```suggestion
    "Data Source: Elastic Endgame",
    "Data Source: Elastic Defend",
```

File: rules_building_block/defense_evasion_services_exe_path.toml [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+name = "Service Path Modification via sc.exe"
+risk_score = 21
+rule_id = "c5677997-f75b-4cda-b830-a75920514096"
+severity = "low"
+tags = [
+    "Domain: Endpoint",
+    "OS: Windows",
+    "Use Case: Threat Detection",
+    "Tactic: Defense Evasion",
+    "Data Source: Elastic Endgame",
```
Comment (w0rk3r) [inline rules_building_block/defense_evasion_services_exe_path.toml:29]:
```suggestion
    "Data Source: Elastic Endgame",
    "Data Source: Elastic Defend",
```

File: rules_building_block/persistence_netsh_helper_dll.toml [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+"""
+from = "now-9m"
+index = ["logs-endpoint.events.*", "endgame-*"]
+language = "eql"
+license = "Elastic License v2"
+name = "Netsh Helper DLL"
+risk_score = 21
+rule_id = "b0638186-4f12-48ac-83d2-47e686d08e82"
+severity = "low"
+tags = ["Domain: Endpoint", "OS: Windows", "Use Case: Threat Detection", "Tactic: Persistence", "Data Source: Elastic Endgame", "Rule Type: BBR"]
```
Comment (w0rk3r) [inline rules_building_block/persistence_netsh_helper_dll.toml:25]:
```suggestion
tags = ["Domain: Endpoint", "OS: Windows", "Use Case: Threat Detection", "Tactic: Persistence", "Data Source: Elastic Endgame", "Data Source: Elastic Defend", "Rule Type: BBR"]
```

File: rules_building_block/persistence_startup_folder_lnk.toml [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+"""
+from = "now-9m"
+index = ["logs-endpoint.events.*", "endgame-*"]
+language = "eql"
+license = "Elastic License v2"
+name = "Shortcut File Written or Modified on Startup Folder"
+risk_score = 21
+rule_id = "ee53d67a-5f0c-423c-a53c-8084ae562b5c"
+severity = "low"
+tags = ["Domain: Endpoint", "OS: Windows", "Use Case: Threat Detection", "Tactic: Persistence", "Data Source: Elastic Endgame", "Rule Type: BBR"]
```
Comment (w0rk3r) [inline rules_building_block/persistence_startup_folder_lnk.toml:24]:
```suggestion
tags = ["Domain: Endpoint", "OS: Windows", "Use Case: Threat Detection", "Tactic: Persistence", "Data Source: Elastic Endgame", "Data Source: Elastic Defend", "Rule Type: BBR"]
```

File: rules_building_block/persistence_werfault_reflectdebugger.toml [hunk 14]
```diff
@@ -0,0 +1,62 @@
...
+"""
+from = "now-9m"
+index = ["logs-endpoint.events.*", "endgame-*"]
+language = "eql"
+license = "Elastic License v2"
+name = "Werfault ReflectDebugger Persistence"
+risk_score = 21
+rule_id = "205b52c4-9c28-4af4-8979-935f3278d61a"
+severity = "low"
+tags = ["Domain: Endpoint", "OS: Windows", "Use Case: Threat Detection", "Tactic: Persistence", "Data Source: Elastic Endgame", "Rule Type: BBR"]
```
Comment (w0rk3r) [inline rules_building_block/persistence_werfault_reflectdebugger.toml:24]:
```suggestion
tags = ["Domain: Endpoint", "OS: Windows", "Use Case: Threat Detection", "Tactic: Persistence", "Data Source: Elastic Endgame", "Data Source: Elastic Defend", "Rule Type: BBR"]
```

**Review by Samirbous — APPROVED** [review]

**Pull request merged**